ams/BaseMigrator.cs

```diff
@@ -63,6 +61,22 @@ protected Task<MediaServicesAccountResource> GetMediaAccountAsync(string mediaAc
                 await Parallel.ForEachAsync(values, options, processItem);
             }
         }
+        protected async Task<(bool, MediaServicesAccountResource?)> IsAMSAccountAsync(string accountName, CancellationToken cancellationToken)
+        {  MediaServicesAccountResource? account = null;
+            try
+            {
+                account = await GetMediaAccountAsync(accountName, cancellationToken);
+                return (true, account);
+            }
+            catch (RequestFailedException ex)
+            {
+                if (ex.ErrorCode != null && ex.ErrorCode.Equals("ResourceNotFound"))
+                {
+                    return (false, account);
+                }
+                return account ==null?(false, account):(true, account);
```

Comment:
This function can be changed as below:

        protected async Task<(bool, MediaServicesAccountResource?)> IsAMSAccountAsync(string accountName, CancellationToken cancellationToken)
        {  
            MediaServicesAccountResource? amsAccount = null;

            try
            {
                amsAccount = await _resourceProvider.GetMediaAccountAsync(accountName, cancellationToken);
            }
            catch (Exception ex)
            {
                if (ex is OutOfMemoryException) throw;  // It is a fatal error.

                // For any other exception, swallow the exception, treat it as not-AMS account, 
                // The caller then has a chance to treat it as storage account and try it again,
                // if it is still failed, the caller will throw exception appropriately.
            }

            return (amsAccount != null, amsAccount);
        }


-------------------------------------------------------------------------------------
Once this function returns true, the AmsAccount must be available, if it is false, the caller should do further check if it is a storage account, otherwise, report error or throw exception in the caller side.